framework/transactions/handlers.py

```diff
@@ -52,7 +53,14 @@ def transaction_after_request(response, base_status_code_error=500):
     if view_has_annotation(NO_AUTO_TRANSACTION_ATTR):
         return response
     if response.status_code >= base_status_code_error:
-        commands.rollback()
+        original_exception = sys.exc_info()
+        try:
+            commands.rollback()
+        except OperationFailure:
+            logger.error('Transaction Rollback failed, original exception was: ', exc_info=original_exception)
```

Comment:
We could just set a property on the response object in `process_exception`, if one isn't already being set, and look at it in `process_response`.